exec_ops/spatialflow_partition/spatialflow_partition.go

```diff
@@ -0,0 +1,302 @@
+
+			// Check if the tile overlaps (consider the buffer) with the ROI (different shapes)
+			handlePoint := func(coordinate []float64) {
+				p := toRelativePixelCoordinate(coordinate)
+				if p.X >= -buffer && p.X <= 1.0 + buffer && p.Y >= -buffer && p.Y <= 1.0 + buffer {
+					isOverlapped = true
+				}  
+			}
+
+			handleLineString := func(coordinates [][]float64) {
+				for _, coordinate := range coordinates {
+					p := toRelativePixelCoordinate(coordinate)
+					if p.X >= -buffer && p.X <= 1.0 + buffer && p.Y >= -buffer && p.Y <= 1.0 + buffer {
+						isOverlapped = true
+					}  
+				}
+				if isOverlapped {
+					return 
+				}
+				for ind, _ := range coordinates {
+					if ind == len(coordinates)-1 {
+						break
+					}
+
+					p1 := toRelativePixelCoordinate(coordinates[ind])
+					p2 := toRelativePixelCoordinate(coordinates[ind+1])
+					
+					segment := gomapinfer.Segment{p1,p2}
+					if segment.Intersection(gomapinfer.Segment{corner1, corner2}) != nil {
+						isOverlapped = true
+						return 
+					}
+					if segment.Intersection(gomapinfer.Segment{corner2, corner3}) != nil {
+						isOverlapped = true
+						return 
+					}
+					if segment.Intersection(gomapinfer.Segment{corner3, corner4}) != nil {
+						isOverlapped = true
+						return 
+					}
+					if segment.Intersection(gomapinfer.Segment{corner4, corner1}) != nil {
+						isOverlapped = true
+						return 
+					}
+				}
+			}
+
+			handlePolygon := func(coordinates [][][]float64) {
+				// We do not support holes yet, so just use coordinates[0].
+				// coordinates[0] is the exterior ring while coordinates[1:] specify
+				// holes in the polygon that should be excluded.
+				handleLineString(coordinates[0])
+
+				if isOverlapped {
+					return
+				}
+				
+				var polygon gomapinfer.Polygon
+				for _, coordinate := range coordinates[0] {
+					p := toRelativePixelCoordinate(coordinate)
+					polygon = append(polygon, p)
+				}
+				
+				for _, corner := range corners {
+					if polygon.Contains(corner) {
+						isOverlapped = true
+						return 
+					}
+				}
+
+			}
+			
+			for _, g := range geometries {
+				if g.Type == geojson.GeometryPoint {
+					handlePoint(g.Point)
+				} else if g.Type == geojson.GeometryMultiPoint {
+					for _, coordinate := range g.MultiPoint {
+						handlePoint(coordinate)
+					}
+				} else if g.Type == geojson.GeometryLineString {
+					handleLineString(g.LineString)
+				} else if g.Type == geojson.GeometryMultiLineString {
+					for _, coordinates := range g.MultiLineString {
+						handleLineString(coordinates)
+					}
+				} else if g.Type == geojson.GeometryPolygon {
+					handlePolygon(g.Polygon)
+				} else if g.Type == geojson.GeometryMultiPolygon {
+					for _, coordinates := range g.MultiPolygon {
+						handlePolygon(coordinates)
+					}
+				}
+
+				if isOverlapped {
+					break
+				}
+			}
+			
+			// If the current tile overlaps with the ROI, store it in a dataset
+			if isOverlapped {
```

Comment:
Usually early return may be more readable especially over time (with more conditions):

	// If the current tile overlaps with the ROI, store it in a dataset
	if !isOverlapped {
		continue
	}
	log.Printf("[spatialflow_partition] create tile %s", fmt.Sprintf("%d_%d_%d", zoom, i, j))
	outputData := skyhook.GeoImageData{
	// ...